internalshared/configutil/config.go

```diff
@@ -98,36 +98,42 @@ func ParseConfig(d string) (*SharedConfig, error) {
 	}
 
 	if o := list.Filter("hsm"); len(o.Items) > 0 {
+		result.found("hsm", "hsm")
 		if err := parseKMS(&result.Seals, o, "hsm", 2); err != nil {
 			return nil, fmt.Errorf("error parsing 'hsm': %w", err)
 		}
 	}
 
 	if o := list.Filter("seal"); len(o.Items) > 0 {
+		result.found("seal", "Seal")
 		if err := parseKMS(&result.Seals, o, "seal", 3); err != nil {
 			return nil, fmt.Errorf("error parsing 'seal': %w", err)
 		}
 	}
 
 	if o := list.Filter("kms"); len(o.Items) > 0 {
+		result.found("kms", "Seal")
```

Comment:
Is this a typo?

Reply:
No, strangely, for the seals, they're all 'seal "sealtype" {' , and we iterate on sealtype but want to remove the "seal" element as one that wasn't found.  TBH it's all a bit of a hack, specifically how HCL 1.0 works with these stanzas.